### PR TITLE
Restore untitled files from within UntitledEditorModel

### LINKED ISSUE #14067: Hot Exit: Restore untitled files from within UntitledEditorModel
I see the motivation for this but it goes through so many layers that we should NOT have this at all. Both untitled and text file world in the end have a good place where you should check for backups and get the contents from there: `TextEditorModel.load()`. That is the only safe place to know if a backup is present and to restore it, everything else is just fragile.

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #14067

### CODE REVIEW

File: src/vs/workbench/services/untitled/common/untitledEditorService.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ export interface IUntitledEditorService {
 	 * It is valid to pass in a file resource. In that case the path will be used as identifier.
 	 * The use case is to be able to create a new file with a specific path with VSCode.
 	 */
-	createOrGet(resource?: URI, modeId?: string, restoreResource?: URI): UntitledEditorInput;
+	createOrGet(resource?: URI, modeId?: string, hasBackupToRestore?: boolean): UntitledEditorInput;
```
Comment (bpasero) [inline src/vs/workbench/services/untitled/common/untitledEditorService.ts]: @Tyriar `hasBackupToRestore` is not being used after all?

File: src/vs/workbench/electron-browser/workbench.ts [hunk 2]
```diff
@@ -298,7 +298,7 @@ export class Workbench implements IPartService {
 				} else {
 					if (this.workbenchParams.options.untitledToRestore && this.workbenchParams.options.untitledToRestore.length) {
 						const untitledToRestoreInputs = this.workbenchParams.options.untitledToRestore.map(resourceInput => {
-							return this.untitledEditorService.createOrGet(null, null, resourceInput.resource);
+							return this.untitledEditorService.createOrGet(resourceInput.resource, null, true);
```
Comment (bpasero) [inline src/vs/workbench/electron-browser/workbench.ts]: @Tyriar Same here, parameter is not being used later?

File: src/vs/workbench/common/editor/untitledEditorInput.ts [hunk 3]
```diff
@@ -87,8 +86,13 @@ export class UntitledEditorInput extends AbstractUntitledEditorInput {
...
 
-		// untitled files with an associated path or restore resource are always dirty
-		return this.hasAssociatedFilePath || !!this.restoreResource;
+		// A disposed input is never dirty, even if it was restored from backup
+		if (this.isDisposed()) {
+			return false;
+		}
+
+		// untitled files with an associated path or associated resource
+		return this.hasAssociatedFilePath || !!this.resource;
```
Comment (bpasero) [inline src/vs/workbench/common/editor/untitledEditorInput.ts]: @Tyriar isn't that wrong? every untitled editor input will have a resource but will not be dirty when opened (e.g. open a new untitled file without content, it should not show dirty).

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/services/untitled/common/untitledEditorService.ts (hunk 1 above)
Comment (Tyriar) [inline src/vs/workbench/services/untitled/common/untitledEditorService.ts]: https://github.com/Microsoft/vscode/pull/15526/commits/647949c0d420854bca0a2d7c479c209d4492d252

File: src/vs/workbench/electron-browser/workbench.ts (hunk 2 above)
Comment (Tyriar) [inline src/vs/workbench/electron-browser/workbench.ts]: https://github.com/Microsoft/vscode/pull/15526/commits/647949c0d420854bca0a2d7c479c209d4492d252

**Pull request merged**